src/lib/tuya.ts

```diff
@@ -855,6 +855,64 @@ export const valueConverter = {
             return {preset: presetLookup[v], system_mode: systemModeLookup[v]};
         },
     },
+    ZWT07_schedule: {
+        from: (value: number[], meta: Fz.Meta, options: KeyValue) => {
+            const programmingMode = [];
+            for (let i = 0; i < 8; i++) {
```

Comment:
This for loop is exactly the same, I propose to move it to a function `parseScheduleProgrammingMode`